### DISCUSSION TITLE
Easiest way to call javascript method in mdx/mdx-deck

### DISCUSSION THREAD

**Discussion by tjtaill** [mdx-js/mdx#1437] [Q&A] [answered]
Hi I want to give a talk on snowplow setup I did I want call make a simple javascript call which sends an event on each page/slide transition. 

I tried
<>{analytics.page()}</> in a slide this successfully sent  the page view event to my server and was recorded but then the page won't render not sure the best way to call analytics.page() on each slide/page is. Maybe create a jsx function that returns some non renderable element and also calls my function ? Or do I have to do all the work of creating a custom provider. 

Any pointers or advice on the simplest way to achieve this  would be greatly appreciated thanks

**Comment by ChristianMurphy** [comment, marked as answer]
You could embed a function in each slide as you outlined above, or you could watch for state/context transitions as noted in https://github.com/jxnblk/mdx-deck/issues/772#issuecomment-758229716
